### PR TITLE
67 introduce builtins module to exec

### LINKED ISSUE #67: Introduce builtins module to exec

### PR DESCRIPTION
**Pull request by alex216**
fix #67 

## Summary by CodeRabbit

- **New Features**
	- Updated built-in commands (`echo`, `pwd`) to utilize a new context structure for improved environment management.
	- Enhanced command execution logic to explicitly handle built-in commands.

- **Bug Fixes**
	- Improved handling of command-line options in the `builtins_echo` function.

- **Tests**
	- Updated test cases for `builtins_echo` and `builtins_pwd` to reflect changes in function parameters.

- **Documentation**
	- Updated author information in multiple files to reflect current contributions.

### CODE REVIEW

**Comment by alex216** [comment]
@rask24  please check the pr!

File: src/exec/execute_command.c [hunk 1]
```diff
@@ -12,7 +12,94 @@
...
+	int	fd_array[3];
+
+	status = 0;
+	if (is_builtin(node->cmd_args->content))
+	{
+		push_io(fd_array);
+		handle_io(conf, node->redirects);
+		status = execute_builtin_command(node, ctx);
+		pop_io(fd_array);
+		return (status);
```
Comment (rask24) [inline src/exec/execute_command.c:101]:
``` suggestion
EXIT_SUCCESS
```

**Review by rask24 — COMMENTED** [review]
@alex216 please check the comments.

**Review by rask24 — APPROVED** [review]
@alex216 LGTM!

**Pull request merged**